app/controllers/users_controller.rb

```diff
@@ -29,10 +29,10 @@ def show
     @tutoring_sessions = TutoringSession.all
 
     # See if there is a spartan session to check into
-    @spartan_session = SpartanSession.where('session_datetime > :now',
+    @spartan_session = SpartanSession.where('session_datetime < :now',
                                             now: Time.zone.now.to_datetime)
-                                     .and(SpartanSession.where('session_datetime < :endTime',
-                                                               endTime: (Time.zone.now + 7200)
+                                     .and(SpartanSession.where('session_datetime > :endTime',
+                                                               endTime: (Time.zone.now - 7200)
```

Comment:
Are you sure? These < and > seem really unintuitive in their placement, considering you're asking it to be less than now and greater than end.

Reply:
The operators were mixed up previously as I found out through testing. The new changes make the check in appear when:

Time.now - 7200 < spartan_session.session_datetime < Time.now

Therefore, the check in is only active for 2 hours after the session starts. Before now, the check in was being calculated such that:

Time.now < spartan_session.session_datetime < Time.now + 7200

This was checking for spartan sessions that start in the next 2 hours.  Which means that a user can't see the session when it starts. This change fixes that bug.

Reply:
Newest push renames the variable to make it easier to read.